aspnet/fundamentals/localization.rst

```diff
@@ -1,8 +1,302 @@
+
+.. code-block:: HTML
+
+  @Localizer["<i>Hello</i> <b>{0}!</b>", UserManager.GetUserName(User)]
+ 
+A French resource file could contain the following:
+
+========================  ===============================  
+Key                       Value    
+========================  ===============================  
+<i>Hello</i> <b>{0}!</b>  <i>Bonjour</i> <b>{0}!</b>  
+========================  ===============================
+
+The rendered view would contain the HTML markup from the resource file.
+
+To use a shared resource file in a view, create a service that exposes ``IHtmlLocalizer``:
+
+.. literalinclude:: localization/sample/Services/LocService.cs
+  :language: c#
+  :linenos:
+  :lines: 5-17
+  :dedent: 3
+
+Register the service:
+
+.. literalinclude:: localization/sample/StartupConfig.cs
+  :language: c#
+  :linenos:
+  :lines: 44-45,55-63
+  :dedent: 6
+  :emphasize-lines: 10,11
+
+Use the service in a view:
+
+.. literalinclude:: localization/sample/Views/Test/About.cshtml
+  :language: HTML
+  :linenos:
+  :emphasize-lines: 5,12
+
+DataAnnotations localization
+------------------------------------
+
+DataAnnotations error messages are localized with ``IStringLocalizer<Type>``. Using the option ``ResourcesPath = "Resources"``, the error messages in ``RegisterViewModel`` can be stored in either of the following paths:
+
+- Resources/ViewModels.Account.RegisterViewModel.fr.resx
+- Resources/ViewModels/Account/RegisterViewModel.fr.resx
+  
+.. literalinclude:: localization/sample/ViewModels/Account/RegisterViewModel.cs
+  :language: c#
+  :lines: 9-26
+  :dedent: 3
+  
+Currently, the run time doesn't lookup localized strings for non-validation attributes. In the code above, the "Email" (from ``[Display(Name = "Email")]``) will not be localized. We hope to add support for this `after RTM <https://github.com/aspnet/Mvc/issues/3518>`_.
+  
+Localization middleware
+-----------------------------------
+
+The current culture on a request is set in the localization :doc:`/fundamentals/middleware`. The localization middleware is enabled in the ``Configure`` method of *Startup.cs* file.
+
+.. literalinclude:: localization/sample/StartupConfig.cs
+  :language: c#
+  :linenos:
+  :lines: 106-107, 136-159
+  :dedent: 6
+
+
+``UseRequestLocalization`` initializes a ``RequestLocalizationMiddleware`` object. The ``Invoke`` method of the `RequestLocalizationMiddleware <https://github.com/aspnet/Localization/blob/dev/src/Microsoft.AspNetCore.Localization/RequestLocalizationMiddleware.cs>`__ class is called by the ASP.NET Core pipeline. The ``Invoke`` method enumerates the ``RequestCultureProviders`` list, and returns the first non-null provider. The default providers come from the ``RequestLocalizationOptions`` class:
```

Comment:
This seems to dive deeply into implementation details. Remove stuff about `Invoke` etc.

Could just say that on every request the list of registered FooProviders in the FooOptions is enumerated and the first non-null provider is used.